plan.go

```diff
@@ -169,14 +180,11 @@ func extractSelectionSet(ctx *PlanningContext, insertionPoint []string, parentTy
 				selection.TypeCondition,
 				selection.SelectionSet,
 				location,
-				childstep,
 			)
 			if err != nil {
 				return nil, nil, err
 			}
-			if !selectionSetHasFieldNamed(selectionSet, "__typename") {
-				selectionSet = append(selectionSet, &ast.Field{Alias: "__typename", Name: "__typename", Definition: &ast.FieldDefinition{Name: "__typename", Type: ast.NamedType("String", nil)}})
-			}
+			selectionSet = append(selectionSet, &ast.Field{Alias: "__typename", Name: "__typename", Definition: &ast.FieldDefinition{Name: "__typename", Type: ast.NamedType("String", nil)}})
```

Comment:
The `selectionSetHasFieldNamed` function was buggy because it didn't account for field aliases. Rather than expanding its implementation to handle more complex naming considerations, I've opted to remove it entirely given that it was adding loops mostly for the sake of vanity selections. A `__typename` field will now _always_ be appended to fragments, even if it was also included in the user-defined selection.